### DISCUSSION TITLE
Hubs in Chrome

### DISCUSSION THREAD

**Discussion by Kanishk-Malick** [mozilla/hubs#2934] [General Questions & Discussion] [answered]
Why doesn't Hubs work well in chrome ?

**Comment by emclaren** [comment]
Hi @Kanishk-Malick - Hubs should work well on Chrome. Can you give more information on what's happening? Also, can you share more about your device (mobile phone /mac computer/ windows computer, operating version)? Does it work on other browsers?

To troubleshoot, I would double check Chrome is up to date, and I would test running Hubs in incognito (private browsing) mode - this would help determine if perhaps there is a browser extension causing issues.

**Reply by Kanishk-Malick** [reply]
Hi @emclaren 
It works the best in Microsoft Edge, then Mozilla Firefox and the worst in Chrome. 
The movement is not at all smooth in chrome. I have made a video, please check here : https://youtu.be/N3hRJ_GObCA
I tried it in incognito mode as well but no use. 
Chrome is up to date. Rest of the system info is mentioned below : 
OS : Windows 10 Home 
Version : 10.0.19041
System Model : Lenovo 80Q3
Type : x64
Processor : Intel(R) Core(TM) i5-6200U CPU @ 2.30GHz, 2400 Mhz, 2 Core(s), 4 Logical Processor(s)
RAM : 8 GB
Available : 1.98 GB

**Comment by vincentfretin** [comment, marked as answer]
The scene you showed in your youtube video is an heavy scene, I experienced 34fps for this scene, see my comments at 
https://www.meetup.com/fr-FR/AWE-Nite-Berlin/events/270732606/comments/509995993/?read=1&_xtd=gatlbWFpbF9jbGlja9oAJDA0ZWViMDcwLTA3NjYtNDc3MC05OWFkLTdjNjc4OTgzYmNiYQ&_af=event&_af_eid=270732606&itemTypeToken=COMMENT

**Reply by Kanishk-Malick** [reply]
If that's the case, it should not work well in other browsers. It is working fine in Microsoft Edge. As a matter of fact, none of the scene work well in chrome for me.

**Reply by Kanishk-Malick** [reply]
@vincentfretin

**Reply by vincentfretin** [reply]
Ok, so it's a different issue not related to the scene. I'm on Ubuntu 18.04 on Dell XPS 13 with Intel graphic card, it works really great on Chrome (85.0.4183.83), and the worst is on Firefox (80.0). With default scene with Chrome alone I'm at 60fps, with Firefox alone I'm at 30fps. If I open both browsers in the same Hubs room, Chrome is at 40fps and Firefox is at 10fps.

For Microsoft Edge is this still Legacy Edge or is new Edge based on Chromium you're using? If it's the one based on Chromium, it's weird that you don't have the same issue on Edge.

Do you have a special anti-virus or firewall? Try to disable it to see if it's the one causing the issue.

**Reply by Kanishk-Malick** [reply]
It is chromium based Edge. No, the antivirus is not causing it.

**Reply by vincentfretin** [reply]
And with https://vrland.io or https://framevr.io do you have the same issue with Chrome? If yes this may be an OS issue, if no this may be js code issue in Hubs in that case you can profile the javascript execution with Chrome.

**Reply by Kanishk-Malick** [reply]
> And with https://vrland.io or https://framevr.io do you have the same issue with Chrome? If yes this may be an OS issue, if no this may be js code issue in Hubs in that case you can profile the javascript execution with Chrome.

Yes, have same issues with both of them. i noticed that these websites are using full body avatars. How ?

**Reply by vincentfretin** [reply]
For the full body avatar on vrland, it's a rigged 3d model, you can create one for Hubs too but the walking animation and sound of the steps when you move are not currently implemented in Hubs.

So for your performance issue, I don't think this a Hubs issue, but a more general issue with your chrome like WebGL rendering? Can you try some simple examples at https://threejs.org/ ?
You can always try running some profiling to see what is taking time, maybe you will have some more cues. https://developers.google.com/web/tools/chrome-devtools/evaluate-performance

**Reply by emclaren** [reply]
Another thing that might be causing this issue is if you have a graphics card installed, perhaps Chrome is for some reason using the integrated graphics card while the other browsers are using the more powerful one. 

One of our community members once shared a link to this support page to ensure that an NVIDIA graphics card was set as the default for their browser (theirs was using the built in card instead by mistake) https://nvidia.custhelp.com/app/answers/detail/a_id/2615/~/how-do-i-customize-optimus-profiles-and-settings%3F

**Reply by emclaren** [reply]
Since this issue seems to be happening with other sites (https://vrland.io & https://framevr.io ) it seems unlikely to be directly related to Hubs. I'm going to mark this as answered for now, but please feel free to follow up with further questions if needed.

**Comment by Kanishk-Malick** [comment]
Hi,
Wrong analysis. None of my web browsers be it chrome, Edge or Firefox are
using Nvidia graphics cards. All of them are operating on integrated
graphics. I even changed the default to Nvidia for chrome but faced the
same issue.

Warm Regards,
Kanishk Malick

**Comment by Kanishk-Malick** [comment]
It is happening in Hubs as well, so please don't mark it answered as of now
please.

Warm Regards,
Kanishk Malick

**Reply by rawnsley** [reply]
If it's happening in vrland.io and framevr.io then it's not a Hubs specific issue. I don't think it's fair to keep the issue open here and I doubt the Hubs team can do much about it, particularly if they are unable to reproduce it.

Your best bet is to raise it upstream with Chrome or work out what component is common to all three and raise it there. In all cases you will need a definitive reproduce that demonstrates the problem.
